test/e2e/placement_selecting_resources_test.go

```diff
@@ -592,3 +594,430 @@ var _ = Describe("validating CRP when deleting resources in a matching namespace
+	})
+
+	It("should remove controller finalizers from CRP", func() {
+		finalizerRemovedActual := allFinalizersExceptForCustomDeletionBlockerRemovedFromCRPActual()
+		Eventually(finalizerRemovedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to remove controller finalizers from CRP %s", crpName)
+	})
+})
+
+var _ = Describe("validating CRP when failed to apply resources", Ordered, func() {
+	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+
+	BeforeAll(func() {
+		By("creating work resources on hub cluster")
+		createWorkResources()
+
+		By("creating work namespace on member cluster")
+		ns := workNamespace()
+
+		Expect(memberCluster1.KubeClient.Create(ctx, &ns)).Should(Succeed(), "Failed to create namespace %s", ns.Name)
+
+		// Create the CRP.
+		crp := &placementv1beta1.ClusterResourcePlacement{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crpName,
+				// Add a custom finalizer; this would allow us to better observe
+				// the behavior of the controllers.
+				Finalizers: []string{customDeletionBlockerFinalizer},
+			},
+			Spec: placementv1beta1.ClusterResourcePlacementSpec{
+				ResourceSelectors: workResourceSelector(),
+			},
+		}
+		By(fmt.Sprintf("creating placement %s", crpName))
+		Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP %s", crpName)
+	})
+
+	AfterAll(func() {
+		By(fmt.Sprintf("deleting placement %s", crpName))
+		cleanupCRP(crpName)
+
+		By("deleting created work resources on hub cluster")
+		cleanupWorkResources()
+
+		By("deleting created work resources on member cluster")
+		cleanWorkResourcesOnCluster(allMemberClusters[0])
+	})
+
+	It("should update CRP status as expected", func() {
+		crpStatusUpdatedActual := func() error {
+			crp := &placementv1beta1.ClusterResourcePlacement{}
+			if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+				return err
+			}
+
+			workNamespaceName := fmt.Sprintf(workNamespaceNameTemplate, GinkgoParallelProcess())
+			appConfigMapName := fmt.Sprintf(appConfigMapNameTemplate, GinkgoParallelProcess())
+			wantStatus := placementv1beta1.ClusterResourcePlacementStatus{
+				Conditions: []metav1.Condition{
+					{
+						Type:               string(placementv1beta1.ClusterResourcePlacementScheduledConditionType),
+						Status:             metav1.ConditionTrue,
+						ObservedGeneration: crp.Generation,
+					},
+					{
+						Type:               string(placementv1beta1.ClusterResourcePlacementSynchronizedConditionType),
+						Status:             metav1.ConditionTrue,
+						ObservedGeneration: crp.Generation,
+					},
+					{
+						Type:               string(placementv1beta1.ClusterResourcePlacementAppliedConditionType),
+						Status:             metav1.ConditionFalse,
+						ObservedGeneration: crp.Generation,
+					},
+				},
+				PlacementStatuses: []placementv1beta1.ResourcePlacementStatus{
+					{
+						ClusterName: memberCluster1Name,
```

Comment:
I don't get why only this cluster apply failed from the diff only (didn't check it out in an IDE).